luarules/gadgets/unit_attached_con_turret_mex.lua

```diff
@@ -0,0 +1,115 @@
+function gadget:GetInfo()
+    return {
+        name      = 'Legion Con Turret Metal Extractor',
+        desc      = 'Allows the mex to function as a con turret by replacing it with a fake mex with a con turret attached',
+        author    = 'EnderRobo',
+        version   = 'v1',
+        date      = 'September 2024',
+        license   = 'GNU GPL, v2 or later',
+        layer     = 12,
+        enabled   = true
+    }
+end
+
+if not gadgetHandler:IsSyncedCode() then
+    return false
+end
+
+local SpGetUnitPosition = Spring.GetUnitPosition
+local SpGetUnitDefID = Spring.GetUnitDefID
+local SpGetUnitAllyTeam = Spring.GetUnitAllyTeam
+local SpGetUnitHealth = Spring.GetUnitHealth
+local SpGetUnitBuildFacing = Spring.GetUnitBuildFacing
+local SpGetUnitIsDead = Spring.GetUnitIsDead
+local SpGetUnitStates = Spring.GetUnitStates
```

Comment:
I believe our stance on storing (caching) functions is to only do it if they get used several times per frame. As this comes at a memory cost.

Reply:
Removed and corrected the script to not use them.